hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/snapshot/OmSnapshotUtils.java

```diff
@@ -137,13 +144,28 @@ public static void createHardLinks(Path dbPath) throws IOException {
     }
   }
 
+  private static boolean isSamePartition(Path fromPathParent, Path toPathParent) throws IOException {
+    try {
+      if (fromPathParent == null || toPathParent == null) {
```

Comment:
This check should be at the start of the method outside of this try catch block for clarity.

Reply:
+1. You can also use `Objects.requireNonNull(fromPathParent, "From path is null.")` and same for `toPathParent`.